src/swap.flows/BTC2ETHTOKEN.js

```diff
@@ -431,7 +434,12 @@ export default (tokenName) => {
             resolve( balance );
           }
           else {
-            setTimeout( checkEthBalance, 20 * 1000 );
+            checkEthBalance()
+            await util.helpers.repeatAsyncUntilResult(() => {
+              if (!this.state.isStoppedSwap) {
+                this.swap.room.once('swap was canceled', () => this.stopSwapProcessParticipant() )
+              }},
```

Comment:
Из этого цикла нужно выходить с помощью return. Например, так: 

```suggestion
              } else {
                return true
              }
            },
```